### PR TITLE
[website] add snackpub a/b testing

### PR DESCRIPTION
**Pull request by Kudo**
# Why

close ENG-7103
close ENG-7107

## pr series

#376
#377
👉 #378 
#379
#380 
#381

# How

- rollout snackpub a/b testing from website. currently 90% pubnub and 10% snackpub + pubnub by default.
- can also add a `testTransport=snackpub` or `testTransport=pubnub` in query string to override the transport from a/b testing randomness.

# Test Plan

- snackpub integration test
- ci passed

### CODE REVIEW

**Review by byCedric — APPROVED** [review]

File: website/deploy/production/snack.env [hunk 1]
```diff
@@ -1,6 +1,7 @@
 LEGACY_SERVER_URL=https://expo.io
 SERVER_URL=https://expo.dev
 API_SERVER_URL=https://exp.host
+SNACKPUB_URL=https://snackpub-3hn4fth2vq-uc.a.run.app
```
Comment (nicknovitski) [inline website/deploy/production/snack.env:4]: These are stable urls?

**Review by nicknovitski — APPROVED** [review]

File: website/deploy/production/snack.env (hunk 1 above)
Comment (Kudo) [reply website/deploy/production/snack.env:4]: it seems so. between deployments and even delete/recreate the cloud run service, it still sticks on the same name. as the url is not visible to users, if it changed, we could still re-deploy snack with new urls. that's why i didn't put the cloud run service behind google cloud load balancer.

Comment (ide) [reply website/deploy/production/snack.env:4]: It doesn't need to block this PR but let's set up a load balancer later

**Review by ide — COMMENTED** [review]
Removing myself as a review; will let others take over for code review of the client-side changes.

**Pull request merged**